src/http-utils.js

```diff
@@ -181,17 +186,48 @@ function getHttpTransferOptions(options, directBinaryUploadOptions) {
         maxConcurrent: directBinaryUploadOptions.getMaxConcurrent(),
     };
 
+    const { protocol = 'http:' } = URL.parse(directBinaryUploadOptions.getUrl());
     const proxyOptions = getProxyAgentOptions(directBinaryUploadOptions);
-    if (proxyOptions) {
-        const { protocol = 'http:' } = URL.parse(directBinaryUploadOptions.getUrl());
-        transferOptions.requestOptions = {
-            agent: protocol === 'https:' ? new HttpsProxyAgent(proxyOptions) : new HttpProxyAgent(proxyOptions)
-        };
+    const proxyUrl = proxyOptions && proxyOptions.href ? proxyOptions.href : undefined;
+
+    const agent = getHttpAgent(protocol, proxyUrl, options.strictSSL);
+    if (agent) {
+      transferOptions.requestOptions = { agent };
```

Comment:
Just as a general suggestion, when I start to see a lot of branching logic in a single function, I'll break code blocks into small functions. I find that it makes writing tests easier, because I can just pass different scenarios into the smaller functions rather than trying to test a bunch of different paths in a single, giant method.

That's my preference, though. But I'd recommend at least giving it a try and seeing what you think.